services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy

```diff
@@ -0,0 +1,158 @@
+package org.openkilda.functionaltests.spec.switches
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.model.MeterId.MAX_SYSTEM_RULE_METER_ID
+import static org.openkilda.model.MeterId.MIN_FLOW_METER_ID
+import static org.openkilda.testing.Constants.NON_EXISTENT_FLOW_ID
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.helpers.SwitchHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.Message
+import org.openkilda.messaging.command.CommandData
+import org.openkilda.messaging.command.CommandMessage
+import org.openkilda.messaging.command.flow.InstallEgressFlow
+import org.openkilda.messaging.command.flow.InstallIngressFlow
+import org.openkilda.messaging.command.flow.InstallTransitFlow
+import org.openkilda.messaging.command.switches.DeleteRulesAction
+import org.openkilda.model.Cookie
+import org.openkilda.model.FlowEncapsulationType
+import org.openkilda.model.OutputVlanType
+
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+
+class SwitchActivationSpec extends HealthCheckSpecification {
+    @Value("#{kafkaTopicsConfig.getSpeakerFlowTopic()}")
+    String flowTopic
+    @Autowired
+    @Qualifier("kafkaProducerProperties")
+    Properties producerProps
+    @Autowired
+    SwitchHelper switchHelper
+
+    def "Missing flow rules are synced from a new switch connected to the controller"() {
```

Comment:
I agree when you say '_Excess_ rules are synced _FROM_...', but _missing_ rules are synced _to_ the switch. Or use 'removed'/'installed' verbs instead.

Reply:
fixed